client/src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import FocusLock from "react-focus-lock";
+
+import "./Sidebar.scss";
+
+import { CSSTransition } from "react-transition-group";
+
+import { trivial_text_maker } from "src/models/text";
+
+import { IconX } from "src/icons/icons";
+
+interface SidebarProps {
+  is_open: boolean;
+  open_close_callback: (value: boolean) => void;
+  children: React.ReactNode;
+  title_text: string;
+  sidebar_toggle_target: string;
+  return_focus_target: HTMLElement | null;
+}
+
+export class Sidebar extends React.Component<SidebarProps> {
+  title = React.createRef<HTMLDivElement>();
+  sidebar_ref = React.createRef<HTMLDivElement>();
+  private key_down = false;
+  constructor(props: SidebarProps) {
+    super(props);
+  }
+
+  componentDidMount() {
+    window.addEventListener("click", this.handleWindowClick, {
+      capture: true,
+    });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("click", this.handleWindowClick, {
+      capture: true,
+    });
+  }
+
+  closeSidebar = (e: Event) => {
+    const sidebar_node = this.sidebar_ref.current;
+    if (
+      this.props.is_open &&
+      sidebar_node &&
+      !sidebar_node.contains(e.target as HTMLElement)
+    ) {
```

Comment:
The logic for determining if the click is outside the sidebar should probably be inside `handleWindowClick`. After that, both `handleWindowClick` and `closeButtonClick` should call `closeSidebar` so that any closing logic is applied consistently.